src/txn/txn_ckpt.c

```diff
@@ -301,30 +301,28 @@ __wt_checkpoint_get_handles(WT_SESSION_IMPL *session, const char *cfg[])
 	 */
 	if (!WT_IS_METADATA(session->dhandle)) {
 		WT_CURSOR *meta_cursor;
-		bool metadata_race;
 
 		WT_ASSERT(session, !F_ISSET(&session->txn, WT_TXN_ERROR));
 		WT_RET(__wt_metadata_cursor(session, &meta_cursor));
 		meta_cursor->set_key(meta_cursor, session->dhandle->name);
 		ret = __wt_curfile_insert_check(meta_cursor);
 		if (ret == WT_ROLLBACK) {
-			metadata_race = true;
 			/*
-			 * Disable this check and assertion for now - it is
-			 * possible that a schema operation with a timestamp in
-			 * the future is in the metadata, but not part of the
-			 * the checkpoint now that checkpoints can be created
-			 * at the stable timestamp.
-			 * See WT-3559 for context on re-adding this assertion.
+			 * If create or drop or any schema operation of a table
+			 * is with in an user transaction then checkpoint can
+			 * see the dhandle before the commit, which will lead
+			 * to the rollback error. We will ignore this dhandle as
+			 * part of this checkpoint by returning from here.
+			 * Also clear the txn error set.
 			 */
-#if 0
-			ret = 0;
-#endif
-		} else
-			metadata_race = false;
+			if (F_ISSET(&session->txn, WT_TXN_ERROR))
+				F_CLR(&session->txn, WT_TXN_ERROR);
```

Comment:
@bvpvamsikrishna We are sitting on merging this because clearing the `WT_TXN_ERROR` flag here is a layering violation and feels risky (i.e: it assumes that the only case it could be set in this code path is the one we care about).

Could you investigate alternative solutions?